nixos/modules/virtualisation/proxmox-image.nix

```
@@ -181,7 +222,19 @@ with lib;
     boot = {
       growPartition = true;
       kernelParams = [ "console=ttyS0" ];
-      loader.grub.device = lib.mkDefault "/dev/vda";
+      loader.grub.device = lib.mkDefault (if hasNoFsPartition then
+        # In this case there is a separate no-fs partition ("/dev/disk/by-partlabel/no-fs" i.e. "/dev/vda2"),
+        # which is used for parts of GRUB2 bootloader. Do not set it as loader.grub.device.
+        # GRUB installation fails, unless the whole disk is selected.
+        "/dev/vda"
+      else if supportBios then
+        "/dev/vda"
+      else
+        "nodev");
+
+      loader.grub.efiSupport = lib.mkDefault supportEfi;
+      loader.grub.efiInstallAsRemovable = lib.mkDefault supportEfi;
```

```suggestion
      loader.grub = {
        efiSupport = lib.mkDefault supportEfi;
        efiInstallAsRemovable = lib.mkDefault supportEfi;
      };
```
I personally like this more but doesn't matter.